### ISSUE TITLE
Google oauth spreadsheet - Newbie Needs Help

### ISSUE THREAD

**Issue by TYLJ** [burnash/gspread#313]
Before i start, please consider me as newborn baby on programming language. I am starting to use python just couple days ago, and trying to connect python to google sheet to manage data(going to do auto generate by time constraint and certain condition, it will be done later). Right now i have difficulties to connect my python to google. Please help. Below are my  syntax:

C:\Users\NVVN>python
Python 3.5.0 (v3.5.0:374f501f4567, Sep 13 2015, 02:27:37) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> #!/usr/bin/python
... import time
>>> import json
>>> import gspread
>>> from oauth2client.client import SignedJwtAssertionCredentials
>>> from subprocess import call
>>> import sys
>>>
>>> ######## Begin gspread setup ########
... #####################################
...
>>> json_key = json.load(open('C:\Users\NVVN\Desktop\xxxxxxxxxx1.json'))
  File "<stdin>", line 1
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 2-3: truncated \UX
XXXXXXX escape
>>> scope = ['https://spreadsheets.google.com/feeds']
>>> credentials = SignedJwtAssertionCredentials(json_key['xxxxxxxx@developer.gserviceaccount.com'], json_key['-----BEGIN PRIVATE KEY-----\xxxxxxxxxxxxxxx\n-----END PRIVATE KEY-----\n'].encode(),
...
... gc = gspread.authorize(credentials)
...
... sht1 = gc.open_by_key('1v23ZmBRCjABo4qT7hz35pfg1FMmDzT5GdGyLmUzCKBs')
  File "<stdin>", line 5
    sht1 = gc.open_by_key('1v23ZmBRCjABo4qT7hz35pfg1FMmDzT5GdGyLmUzCKBs')
       ^
SyntaxError: invalid syntax
>>>
>>>

**Comment by TYLJ** [comment]
Update:

C:\Users\NVVN>python
Python 3.5.0 (v3.5.0:374f501f4567, Sep 13 2015, 02:27:37) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> #!/usr/bin/python
... import time
>>> import json
>>> import gspread
>>> from oauth2client.client import SignedJwtAssertionCredentials
>>> from subprocess import call
>>> import sys
>>>
>>> ######## Begin gspread setup ########
... #####################################
...
>>> json_key = json.load(open('testxxxxxx.json'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: 'testxxxxxxxx.json'
>>> scope = ['https://spreadsheets.google.com/feeds']
>>> credentials = SignedJwtAssertionCredentials(json_key['xxxxxxxxxxxx.gserviceaccount.com'], json_key['-----BEGIN PRIVATE KEY-----\nxxxxxxxxxxxx\n'].encode(),
...
... gc = gspread.authorize(credentials)
...
... sht1 = gc.open_by_key('xxxxxxxs')
  File "<stdin>", line 5
    sht1 = gc.open_by_key('xxxxxxxs')
       ^
SyntaxError: invalid syntax
>>>

**Comment by msuozzo** [comment]
In Python, backslashes are what's called an escape character. The string `'a \new car'` is parsed as the characters `['a', ' ', '\n', 'e', 'w', ' ', 'c', 'a', 'r']` because the backslash told python "combine the next character with me and we'll mean something together". In python, the `'\n'` character means "newline". Try printing `'a \new car'` and you'll notice `a` on the first line and `ew car` on another.

Unfortunately (or fortunately) for you, the windows file path you typed in coincidentally contains a python escape sequence (if you're so inclined, you can read about this escape sequence [here](https://docs.python.org/2/howto/unicode.html#the-unicode-type)). I say fortunately because the exception immediately alerted you to the line in which your error occurred.

Luckily, python provides an escape sequence just for when you want to type an actual backslash character (imagine that). Remember that the first backslash means "watch this next character"? Well all you need to do to type a literal backslash is type another backslash. So the first example should be `'a \\new car'` which will be parsed as `['a', ' ', '\\', 'n', 'e', 'w', ' ', 'c', 'a', 'r']`

**tl;dr** replace your backslash characters with two backslash characters: `'C:\\Users\\NVVN\\Desktop\\xxxxxxxxxx1.json'`
